fluster/test.py

```diff
@@ -94,14 +96,46 @@ def _test(self) -> None:
             os.remove(output_filepath)
 
         if not self.reference:
-            self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.FAIL
-            if self.test_vector.result.lower() == result.lower():
-                self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.SUCCESS
-            self.assertEqual(
-                self.test_vector.result.lower(),
-                result.lower(),
-                self.test_vector.name,
-            )
+            self.compare_result(result, output_filepath, input_filepath)
         else:
             self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.REFERENCE
             self.test_suite.test_vectors[self.test_vector.name].result = result
+
+    @abstractmethod
+    def compare_result(self, result: str, output_filepath: str, input_filepath: str) -> None:
+        """Compare the test result with the expected value"""
+
+
+class MD5ComparisonTest(Test):
+    """Test class for MD5 comparison"""
+
+    def compare_result(self, result: str, output_filepath: str, input_filepath: str) -> None:
+        self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.FAIL
+        if self.test_vector.result.lower() == result.lower():
+            self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.SUCCESS
+        self.assertEqual(
+            self.test_vector.result.lower(),
+            result.lower(),
+            self.test_vector.name,
+        )
+
+
+class PixelComparisonTest(Test):
+    """Test class for pixel comparison"""
+
+    def compare_result(self, result: str, output_filepath: str, input_filepath: str) -> None:
+        self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.FAIL
+        reference_yuv = ISOMPEG2VDecoder().decode(
```

Comment:
This need to be run in a similar way as the other decoding process, checking for timeouts and exceptions in the decoding process.